### PR TITLE
Restructuring of the hierarchy of useful functions in the RPC crate

### PR DESCRIPTION
**Pull request by jbcaron**
# Restructuring of the hierarchy of useful functions in the RPC crate

# Pull Request type

- Refactoring (no functional changes, no API changes)

## What is the new behavior?

- removing duplicate functions
- restructuring of the files hierarchy
- remove blockifier from `runtime_api`

## Does this introduce a breaking change?

No

### CODE REVIEW

**Comment by jbcaron** [comment]
~🚧 State root doesn't match~ (solve)

**Review by antiyro — APPROVED** [review]

File: crates/client/rpc/src/methods/read/get_class_at.rs [hunk 1]
```diff
@@ -43,8 +43,11 @@ pub fn get_class_at(block_id: BlockId, contract_address: FieldElement) -> RpcRes
     };
 
     // converting from stored Blockifier class to rpc class
-    Ok(ContractClassWrapper { contract: contract_class, abi: contract_abi }.try_into().map_err(|e| {
-        log::error!("Failed to convert contract class at address '{contract_address}' to RPC contract class: {e}");
-        StarknetRpcApiError::InternalServerError
-    })?)
+    // TODO: retrieve sierra_program_length and abi_length when they are stored in the storage
+    Ok(ContractClassWrapper { contract: contract_class, abi: contract_abi, sierra_program_length: 0, abi_length: 0 }
```
Comment (Trantorian1) [inline crates/client/rpc/src/methods/read/get_class_at.rs:47]: Were do we need to get the `sierra_program_length` and `abi_length` from? isn't it already a part of `StorageContactClassData` ?

File: crates/client/rpc/src/utils/execution.rs [hunk 2]
```diff
@@ -0,0 +1,272 @@
...
+        }
+        Err(e) => {
+            log::error!("failed to call smart contract {:?}", e);
+            Err(())
+        }
+    }
+}
+
+#[allow(dead_code)]
+pub fn simulate_message(
```
Comment (Trantorian1) [inline crates/client/rpc/src/utils/execution.rs]: Is it pertinent to keep this function if we are mo longer using it?

File: crates/client/rpc/src/utils/helpers.rs [hunk 3]
```diff
@@ -353,21 +228,9 @@ fn to_legacy_entry_point(entry_point: EntryPoint) -> Result<LegacyContractEntryP
...
-        client.header(block_hash).ok().flatten().ok_or_else(|| anyhow::Error::msg("Failed to retrieve header"))?;
-    let digest = header.digest();
-    let block = find_starknet_block(digest)?;
-    Ok(block)
-}
-
 // Utils to convert Casm contract class to Compiled class
-pub fn get_casm_cotract_class_hash(casm_contract_class: &CasmContractClass) -> FieldElement {
+#[allow(dead_code)]
+pub fn get_casm_contract_class_hash(casm_contract_class: &CasmContractClass) -> FieldElement {
```
Comment (Trantorian1) [inline crates/client/rpc/src/utils/helpers.rs]: Same here, do we still need this function?

**Review by Tbelleng — APPROVED** [review]

File: crates/client/rpc/src/methods/read/get_class_at.rs (hunk 1 above)
Comment (jbcaron) [reply crates/client/rpc/src/methods/read/get_class_at.rs:47]: not the abi

**Pull request merged**